File: solutions/fortran/hamming/1/hamming.f90
module hamming
  implicit none
contains

  function compute(strand1, strand2, distance)
      character(*) :: strand1, strand2
      integer :: distance
      logical :: compute
      integer :: i

      distance = 0
      if (len(strand1) /= len(strand2)) then 
        compute = .false.
      else 
        compute = .true.
        if (len(strand1) > 0) then
         do i = 1,len(strand1)
            if (strand1(i:i) /= strand2(i:i)) then
              distance = distance + 1
            end if
          end do
        end if
      end if

  end function compute

end module hamming

File: solutions/r/hamming/1/hamming.R
# This is a stub function to take two strings
# and calculate the hamming distance
hamming <- function(strand1, strand2) {
  stopifnot(nchar(strand1) == nchar(strand2))
  if(strand1=="") return(0)
  s1 <- strsplit(strand1,"")[[1]]
  s2 <- strsplit(strand2,"")[[1]]
  len <- length(s1)
  dist <- 0
  for(i in 1:len){
      n1 <- s1[i]
      n2 <- s2[i]
      if(n1!= n2){  
        dist <- dist+1 }
    }
  dist
  }
